### PR TITLE
update logger to eth logger

### PR DESCRIPTION
**Pull request by bhakiyakalimuthu**
- Update current logger to eth logger
Questions to be clarified ?
- [x] 1. Need to agree on key value pattern (camelCase or snake_case) ?
- [x] 2. Whether to include method name ?
- [x] 3. Logger message to start with upper or lower case ?

### CODE REVIEW

File: server/redisstate.go [hunk 1]
```diff
@@ -15,27 +15,27 @@ var RedisPrefix = "rpc-endpoint:"
...
 
 // Enable lookup of txFrom by txHash
 var RedisPrefixSenderOfTxHash = RedisPrefix + "txsender-of-txhash:"
-var RedisExpirySenderOfTxHash = time.Duration(24 * time.Hour) // 1 day
+var RedisExpirySenderOfTxHash = 24 * time.Hour // 1 day
 
 // Remember nonce of pending user tx
 var RedisPrefixSenderMaxNonce = RedisPrefix + "txsender-pending-max-nonce:"
-var RedisExpirySenderMaxNonce = time.Duration(2 * time.Hour)
+var RedisExpirySenderMaxNonce = 24 * time.Hour // 1 day
```
Comment (bhakiyakalimuthu) [inline server/redisstate.go:34]: fix this

**Comment by metachris** [comment]
I like it 👍 

* casing maybe just like the variables
* method name must not always be included
* no strong preference about starting upper or lower case

File: server/request_handler.go [hunk 2]
```diff
@@ -38,7 +40,7 @@ func (r *RpcRequestHandler) process() {
 	// At end of request, log the time it needed
 	defer func() {
 		timeRequestNeeded := time.Since(r.timeStarted)
-		r.logger.log("request took %.6f sec", timeRequestNeeded.Seconds())
+		log.Info("[process] Time taken to process request", "request took", fmt.Sprintf("%.6f sec", timeRequestNeeded.Seconds()))
```
Comment (metachris) [inline server/request_handler.go:43]:
```suggestion
		log.Info("[process] Time taken to process request", "sec", timeRequestNeeded.Seconds())
```

Better to keep the values unformatted, and to have only a single word as key

Comment (bhakiyakalimuthu) [reply server/request_handler.go]: Updated as  suggested

File: server/request_handler.go [hunk 3]
```diff
@@ -115,7 +117,7 @@ func (r *RpcRequestHandler) processBatchRequest(jsonBatchReq []*types.JsonRpcReq
 		// Scatter worker
 		go func(count int, rpcReq *types.JsonRpcRequest) {
 			// Create child logger
-			l := r.logger.CreateChildLogger(strconv.FormatInt(int64(count), 10))
+			l := r.logger.New(log.Ctx{"worker-id": strconv.FormatInt(int64(count), 10)})
```
Comment (metachris) [inline server/request_handler.go:120]:
```suggestion
			l := log.New(log.Ctx{"uid": fmt.Sprintf("%s.%d", r.uid, count)})
```

File: server/request_intercepts.go [hunk 4]
```diff
@@ -147,6 +147,6 @@ func (r *RpcRequest) intercept_eth_call_to_FlashRPC_Contract() (requestFinished
 	}
 
 	r.writeRpcResult("0x0000000000000000000000000000000000000000000000000000000000000001")
-	r.logger.log("Intercepted eth_call to FlashRPC contract")
+	r.logger.Info("[call_to_FlashRPC_Contract] Intercepted eth_call to FlashRPC contract")
```
Comment (metachris) [inline server/request_intercepts.go:150]:
```suggestion
	r.logger.Info("Intercepted eth_call to FlashRPC contract")
```

File: server/request_processor.go [hunk 5]
```diff
@@ -49,7 +51,7 @@ func NewRpcRequest(logger Logger, jsonReq *types.JsonRpcRequest, defaultProxyUrl
 }
 
 func (r *RpcRequest) ProcessRequest() *types.JsonRpcResponse {
-	r.logger.log("JSON-RPC request from ip: %s - method: %s / goroutines: %d", r.ip, r.jsonReq.Method, runtime.NumGoroutine())
+	r.logger.Info(fmt.Sprintf("JSON-RPC request from ip: %s - method: %s / goroutines: %d", r.ip, r.jsonReq.Method, runtime.NumGoroutine()))
```
Comment (metachris) [inline server/request_processor.go:54]:
```suggestion
	r.logger.Info("JSON-RPC request", "ip", r.ip, "method", r.jsonReq.Method, "goroutines", runtime.NumGoroutine())
```

File: server/request_processor.go [hunk 6]
```diff
@@ -108,21 +110,20 @@ func (r *RpcRequest) proxyRequestRead(proxyUrl string) (readJsonRpsResponseSucce
 
 	// Afterwards, check time and result
 	timeProxyNeeded := time.Since(timeProxyStart)
-	r.logger.log("proxy response %d after %.6f sec", proxyResp.StatusCode, timeProxyNeeded.Seconds())
-	// r.logger.log("proxy response %d after %.6f: %v", proxyResp.StatusCode, timeProxyNeeded.Seconds(), proxyResp)
+	r.logger.Info(fmt.Sprintf("[proxyRequestRead] proxy response %d after %.6f sec", proxyResp.StatusCode, timeProxyNeeded.Seconds()))
```
Comment (metachris) [inline server/request_processor.go:113]:
```suggestion
	r.logger.Info("[proxyRequestRead] proxied response", "statusCode", proxyResp.StatusCode, "secNeeded", timeProxyNeeded.Seconds())
```

File: server/request_processor.go [hunk 7]
```diff
@@ -166,17 +167,17 @@ func (r *RpcRequest) sendTxToRelay() {
...
 	// Check if tx was already forwarded and should be blocked now
 	if r.blockResendingTxToRelay(txHash) {
-		r.logger.log("[sendTxToRelay] blocked %s", txHash)
+		r.logger.Info("[sendTxToRelay] Blocked", "tx", txHash)
 		r.writeRpcResult(txHash)
 		return
 	}
 
-	r.logger.log("[sendTxToRelay] sending %s -- from ip: %s / address: %s / to: %s", txHash, r.ip, r.txFrom, r.tx.To())
+	r.logger.Info(fmt.Sprintf("[sendTxToRelay] sending %s -- from ip: %s / address: %s / to: %s", txHash, r.ip, r.txFrom, r.tx.To()))
```
Comment (metachris) [inline server/request_processor.go:175]: please log variables separately here (no `fmt.Sprintf`) - this makes it easier to query the logs

File: server/request_processor.go [hunk 8]
```diff
@@ -202,26 +203,26 @@ func (r *RpcRequest) sendTxToRelay() {
 	// https://github.com/ethereum/go-ethereum/blob/master/core/tx_pool.go#L53
 	if r.tx.Size() > 131072 {
 		if _, found := allowedLargeTxTargets[txTo.Hex()]; !found {
-			r.logger.logError("sendTxToRelay] large tx to not allowed target - hash: %s - target: %s", txHash, txTo)
+			r.logger.Error(fmt.Sprintf("[sendTxToRelay] large tx to not allowed target - hash: %s - target: %s", txHash, txTo))
```
Comment (metachris) [inline server/request_processor.go:206]: log variables separately

File: server/request_processor.go [hunk 9]
```diff
@@ -202,26 +203,26 @@ func (r *RpcRequest) sendTxToRelay() {
...
 			r.writeRpcError("invalid target for large tx", types.JsonRpcInternalError)
 			return
 		}
-		r.logger.log("sendTxToRelay] allowed large tx - hash: %s - target: %s", txHash, txTo)
+		r.logger.Info(fmt.Sprintf("sendTxToRelay] allowed large tx - hash: %s - target: %s", txHash, txTo))
```
Comment (metachris) [inline server/request_processor.go:210]: log variables separately

Comment (bhakiyakalimuthu) [reply server/request_processor.go]: updated

File: server/request_processor.go (hunk 8 above)
Comment (bhakiyakalimuthu) [reply server/request_processor.go]: updated

File: server/request_processor.go (hunk 7 above)
Comment (bhakiyakalimuthu) [reply server/request_processor.go]: updated

File: server/request_processor.go (hunk 6 above)
Comment (bhakiyakalimuthu) [reply server/request_processor.go]: updated

File: server/request_intercepts.go (hunk 4 above)
Comment (bhakiyakalimuthu) [reply server/request_intercepts.go]: updated

File: server/request_handler.go (hunk 3 above)
Comment (bhakiyakalimuthu) [reply server/request_handler.go]: updated

**Pull request merged**